backend/services/office_hours/ticket.py

```diff
@@ -1,689 +1,280 @@
+        user_members = self._session.scalars(user_member_query).unique().all()
 
-        # Fetch Office Hours Section From Event Model
-        oh_section = oh_event_model.oh_section
+        user_member_ids = [user_member.user_id for user_member in user_members]
 
-        # Return section model
-        return oh_section
-
-    def _check_ticket_creators_permissions(
-        self, subject: User, oh_ticket: OfficeHoursTicketDraft
-    ) -> list[SectionMemberEntity]:
-        """
-        Check permissions for ticket creators in the context of an office hours ticket.
-
-        Args:
-            subject (User): The user initiating the action.
-            oh_ticket (OfficeHoursTicketDraft): The office hours ticket draft.
-
-        Returns:
-            list[SectionMemberEntity]: A list of section member entities representing
-                the creators of the ticket who have permissions.
+        for creator_id in creator_ids:
+            if creator_id not in user_member_ids:
+                raise CoursePermissionException(
+                    "Not allowed to create a ticket if you are not in the course."
+                )
 
-        Raises:
-            PermissionError: If not all creators of the ticket are members of the current
-                office hours section.
-        """
+        for user_member in user_members:
+            # If the user is not a member of the looked up course, throw an error
+            if not user_member or user_member.member_role != RosterRole.STUDENT:
+                raise CoursePermissionException(
+                    "Not allowed to create a ticket if you are not a student."
+                )
 
-        # OH Section Needed To Fetch Section Member
-        oh_section: OfficeHoursSection = self._get_office_hours_section_by_oh_event_id(
-            oh_ticket.oh_event.id
-        )
+        # Create entity
+        oh_ticket_entity = OfficeHoursTicketEntity.from_new_model(ticket)
 
-        # Build List of Creator IDs - Subject and Creators List (For Group)
-        ticket_creator_ids = [subject.id] + [
-            creator.id for creator in oh_ticket.creators if creator.id != subject.id
-        ]
+        # Add new object to table and commit changes
+        self._session.add(oh_ticket_entity)
 
-        creator_section_member_entities = (
-            self._session.query(SectionMemberEntity)
-            .filter(
-                OfficeHoursSectionEntity.id == oh_section.id,
-            )
-            .filter(SectionEntity.office_hours_id == OfficeHoursSectionEntity.id)
-            .filter(
-                SectionMemberEntity.section_id == SectionEntity.id,
-                SectionMemberEntity.member_role == RosterRole.STUDENT,
-                SectionMemberEntity.user_id.in_(ticket_creator_ids),
-            )
-            .distinct()
-            .all()
-        )
+        # Commit so can get ticket id
+        self._session.commit()
```

Comment:
I believe we should be able to do this all in a single transaction without committing first, then adding later. SQLAlchemy should be able to sync this.

Reply:
@KrisJordan removing the commit does fail - I think this is because the `id` field is not filled in with any value until it has been persisted in the db.